pkg/git/repo.go

```diff
@@ -41,10 +43,11 @@ func (err NotReadyError) Error() string {
 type GitRepoStatus string
 
 const (
-	RepoNoConfig GitRepoStatus = "unconfigured" // configuration is empty
-	RepoNew      GitRepoStatus = "new"          // no attempt made to clone it yet
-	RepoCloned   GitRepoStatus = "cloned"       // has been read (cloned); no attempt made to write
-	RepoReady    GitRepoStatus = "ready"        // has been written to, so ready to sync
+	RepoNoConfig    GitRepoStatus = "unconfigured" // configuration is empty
+	RepoNew         GitRepoStatus = "new"          // no attempt made to clone it yet
+	RepoCloned      GitRepoStatus = "cloned"       // has been read (cloned); no attempt made to write
+	RepoReady       GitRepoStatus = "ready"        // has been written to, so ready to sync
+	RepoUnreachable GitRepoStatus = "unreachable"  // git repo URL is invalid
```

Comment:
For the network error, isn't it the case of `timeout`?

Reply:
If you have some DNS issue, it is possible that Git throws the following error (while the domain is valid):

```
Could not resolve hostname bitbucket.org: Temporary failure in name resolution fatal: Could not read from remote repository.
```

Reply:
What I expect is that even the failure is because of incorrect git URL or DNS issue, it will be in this case. But I got your point, I think we can make the error like `can't connect git repo with url %s, please check your DNS or repository url`.
Where do you stand for this, @hiddeco ?

Reply:
I think the error message _as is_ is fine, my comment was just about the `// git repo URL is invalid` not being in line with what the status actually represents.

I do however wonder if it would help if we would still include the original error (as given by Git) in the `can't connect git repo with url %s` error message, as it will likely help people debugging the issue.

Reply:
The error msg will be looked like this.
`can't connect git repo with url ssh://git@github:someaccount/somerepo, please check your DNS or repository url`.
So I would like to say that origin git error will not be present here.

Reply:
I think that only captures a small part of what can cause this error to surface, while including the original error and a supportive message about the fact that it can't connect to the Git repository would give someone everything they need to be able to debug the issue and sort out what goes wrong.

Reply:
Ok, I will include the origin git error followed by the above msg.